### DISCUSSION TITLE
Foreign Keys

### DISCUSSION THREAD

**Discussion by tim-coder** [cachapa/sql_crdt#10] [General]
I noticed that none of the examples use foreign key constraints. Are their any issues with using FKs with sql_crdt?

**Comment by cachapa** [comment]
To be honest I haven't tested it yet since I don't use FKs in my projects, but I suspect it should work fine.

One possible issue is when merging data, if a record with a FK is imported before the corresponding record in the associated table is merged, though I believe the records are applied in the same order as they were created in the source so that would work well.

Obviously you'll have to make sure that all those tables and records are synchronized.

If you're willing to contribute some code, it would be great to have unit tests for those cases.

**Comment by tim-coder** [comment]
That's what I was thinking - that as long as the order is maintained, there shouldn't be an issue. Thanks so much for the response.

If I get something working, I'll be happy to contribute unit and integration tests. I need to get a better handle on CRDT and your excellent library first.
